pkg/state/snapshot_generator_internal_test.go

```diff
@@ -1048,18 +1130,245 @@ func TestDefaultInvokeScriptSnapshot(t *testing.T) {
+			idxDataSnapshot = idx
+			transactionDataEntrySnapshot = *dataEntryS
+		}
+	}
+	var snapshotI []byte
+	var snapshotJ []byte
+	sort.Slice(expectedDataEntrySnapshot.DataEntries, func(i, j int) bool {
+		snapshotI, err = json.Marshal(expectedDataEntrySnapshot.DataEntries[i])
+		assert.NoError(t, err, "failed to marshal snapshots")
+		snapshotJ, err = json.Marshal(expectedDataEntrySnapshot.DataEntries[j])
+		assert.NoError(t, err, "failed to marshal snapshots")
+		return string(snapshotI) < string(snapshotJ)
+	})
+
+	sort.Slice(transactionDataEntrySnapshot.DataEntries, func(i, j int) bool {
+		snapshotI, err = json.Marshal(transactionDataEntrySnapshot.DataEntries[i])
+		assert.NoError(t, err, "failed to marshal snapshots")
+		snapshotJ, err = json.Marshal(transactionDataEntrySnapshot.DataEntries[j])
+		assert.NoError(t, err, "failed to marshal snapshots")
+		return string(snapshotI) < string(snapshotJ)
+	})
+
+	assert.Equal(t, expectedDataEntrySnapshot, transactionDataEntrySnapshot)
+
+	expectedSnapshot = remove(expectedSnapshot, idxExpectedDataSnapshot)
+	transactionSnapshot = remove(transactionSnapshot, idxDataSnapshot)
+
+	sort.Slice(expectedSnapshot, func(i, j int) bool {
+		snapshotI, err = json.Marshal(expectedSnapshot[i])
+		assert.NoError(t, err, "failed to marshal snapshots")
+		snapshotJ, err = json.Marshal(expectedSnapshot[j])
+		assert.NoError(t, err, "failed to marshal snapshots")
+		return string(snapshotI) < string(snapshotJ)
+	})
+
+	sort.Slice(transactionSnapshot, func(i, j int) bool {
+		snapshotI, err = json.Marshal(transactionSnapshot[i])
+		assert.NoError(t, err, "failed to marshal snapshots")
+		snapshotJ, err = json.Marshal(transactionSnapshot[j])
+		assert.NoError(t, err, "failed to marshal snapshots")
+		return string(snapshotI) < string(snapshotJ)
+	})
+
+	assert.Equal(t, expectedSnapshot, transactionSnapshot)
+	to.stor.flush(t)
+}
+
+func TestDefaultInvokeScriptRepeatableActionsSnapshot(t *testing.T) {
+	/*
+			{-# STDLIB_VERSION 5 #-}
+			{-# CONTENT_TYPE DAPP #-}
+			{-# SCRIPT_TYPE ACCOUNT #-}
+
+			@Callable(i)
+			func call() = {
+			  [
+				BooleanEntry("bool", true),
+				IntegerEntry("int", 1),
+		        IntegerEntry("int", 1),
+		        IntegerEntry("int2", 2),
+				StringEntry("str", "1"),
+		        StringEntry("str", "1"),
+		        StringEntry("str2", "2")
+			  ]
+			}
+	*/
+	script := "AAIFAAAAAAAAAAQIAhIAAAAAAAAAAAEAAAABaQEAAAAEY2Fsb" +
+		"AAAAAAJAARMAAAAAgkBAAAADEJvb2xlYW5FbnRyeQAAAAICAAAABGJvb2wGCQAE" +
+		"TAAAAAIJAQAAAAxJbnRlZ2VyRW50cnkAAAACAgAAAANpbnQAAAAAAAAAAAEJAARMAAAA" +
+		"AgkBAAAADEludGVnZXJFbnRyeQAAAAICAAAAA2ludAAAAAAAAAAAAQkABEwAAAACCQEAAA" +
+		"AMSW50ZWdlckVudHJ5AAAAAgIAAAAEaW50MgAAAAAAAAAAAgkABEwAAAACCQEAAAALU3RyaW5nRW" +
+		"50cnkAAAACAgAAAANzdHICAAAAATEJAARMAAAAAgkBAAAAC1N0cmluZ0VudHJ5AAAAAgIAAAADc3RyAgAAAA" +
+		"ExCQAETAAAAAIJAQAAAAtTdHJpbmdFbnRyeQAAAAICAAAABHN0cjICAAAAATIFAAAAA25pbAAAAACkN9Gf"
+	scriptsBytes, err := base64.StdEncoding.DecodeString(script)
```

Comment:
We don't have to store compiled code here. Just use the compiler package and it will create parsed tree from ride script source code.